src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -536,33 +534,40 @@ class GroupReadsByUmi
           r(this.assignTag) = truncated
         }
 
-        sorter += r
         kept += 1
-        sortProgress.record(r)
+        r
       }
 
-    logger.info(f"Accepted $kept%,d reads for grouping.")
-    if (filteredNonPf > 0) logger.info(f"Filtered out $filteredNonPf%,d non-PF reads.")
-    logger.info(f"Filtered out $filteredPoorAlignment%,d reads due to mapping issues.")
-    logger.info(f"Filtered out $filteredNsInUmi%,d reads that contained one or more Ns in their UMIs.")
-    this.minUmiLength.foreach { _ => logger.info(f"Filtered out $filterUmisTooShort%,d reads that contained UMIs that were too short.") }
-
-    // Output the reads in the new ordering
-    logger.info("Assigning reads to UMIs and outputting.")
     val outHeader = header.clone()
-    SamOrder.TemplateCoordinate.applyTo(outHeader)
-    val out = SamWriter(output, outHeader)
-
-    val iterator = Bams.templateIterator(sorter.iterator, out.header, Bams.MaxInMemory, Io.tmpDir)
     val tagFamilySizeCounter = new NumericCounter[Int]()
```

Comment:
It's weird that after your edit this is sandwiched between creation of the outHeader and setting the SamOrder on the outHeader.  Maybe move it up a line?